proto/pocket/service/relay.proto

```diff
@@ -9,50 +9,88 @@ import "gogoproto/gogo.proto";
+  // See: https://github.com/pokt-network/poktroll/pull/567#discussion_r1628722168
   //
-  // The supplier operator address the relay is sent to. It is being used on the
-  // RelayMiner to route to the correct supplier.
-  string supplier_operator_address = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // The Bech32 address of the application.
+  // Supplier operator address:
+  // - The Bech32 address of the supplier operator the relay is sent to.
+  // - Used by the RelayMiner to route to the correct supplier.
+  string supplier_operator_address = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
 }
 
-// RelayRequest holds the request details for a relay.
+// RelayRequest
+//
+// Holds the request details for a relay.
 message RelayRequest {
   RelayRequestMetadata meta = 1 [(gogoproto.nullable) = false];
-  // payload is the serialized payload for the request.
-  // The payload is passed directly to the service and as such can be any
-  // format that the service supports: JSON-RPC, REST, gRPC, etc.
+
+  // Serialized request payload:
+  // - Passed directly to the service.
+  // - Can be any supported format: JSON-RPC, REST, gRPC, etc.
   bytes payload = 2;
 }
 
-// RelayResponse contains the response details for a RelayRequest.
+// RelayResponse
+//
+// Contains the response details for a RelayRequest.
 message RelayResponse {
   RelayResponseMetadata meta = 1 [(gogoproto.nullable) = false];
-  // payload is the serialized payload for the response.
-  // The payload is passed directly from the service and as such can be any
-  // format the service responds with: JSON-RPC, REST, gRPC, etc.
+
+  // Serialized response payload:
+  // - Passed directly from the service.
+  // - Can be any supported format: JSON-RPC, REST, gRPC, etc.
   bytes payload = 2;
+
+  // Error returned by the RelayMiner, if applicable.
+  // - If no error occurred, this field is empty.
+  RelayMinerError relay_miner_error = 3;
 }
 
-// RelayResponseMetadata contains the metadata for a RelayResponse.
+// RelayResponseMetadata
+//
+// Contains metadata for a RelayResponse.
 message RelayResponseMetadata {
-  session.SessionHeader session_header = 1; // Session header associated with the relay.
-  bytes supplier_operator_signature = 2; // Signature of the supplier's operator on the response.
+  // Session header associated with the relay.
+  session.SessionHeader session_header = 1;
+
+  // Signature of the supplier's operator on the response.
+  bytes supplier_operator_signature = 2;
 }
+
+// RelayMinerError
+//
+// Contains error details returned by the RelayMiner.
+message RelayMinerError {
+  // Registered codespace for the error (groups errors by source/module).
+  string codespace = 1;
+
+  // Specific registered error code.
+  uint32 code = 2;
+
+  // Human-readable, concise error description.
+  string description = 3;
+
+  // Detailed error message (may include additional context).
+  string message = 4;
```

Comment:
It's unclear to me what the distinction between these two should be. 

Consider consolidating them into one.

Reply:
I think we should keep both.
See the updated comments.